File: lib/logger.cpp
#include <assert.h>
#include <logger.h>

using namespace SEP;
void buffer::add(const std::string &line) {
  _buf.push_back(line);
  if (_buf.size() == _flush) flush();
  _buf.resize(0);
}
fileBuffer::fileBuffer(const std::string fle, const int flushsz) {
  _flush = flushsz;
  _myf.open(fle);
}

void logger::addMessage(const std::string &buf, const std::string &msg) {
  getBuffer(buf)->add(msg);
}
std::shared_ptr<buffer> logger::getBuffer(const std::string &buf) {
  if (_bufs.count(buf) == 0) _bufs[buf].reset(new errBuffer(100));
  return _bufs[buf];
}
void logger::createFileBuffer(const std::string &buf, const std::string &fle,
                              const int nflush) {
  _bufs[buf].reset(new fileBuffer(fle, nflush));
}
void logger::createErrBuffer(const std::string &buf, const int nflush) {
  _bufs[buf].reset(new errBuffer(nflush));
}
/*
void logMessage(std::&msg) { logger::get()->addMessage("default", msg); }
void logBuffer(std::string &buf, std::string &msg) {
  logger::get()->addMessage(buf, msg);
}
std::string getBuffer(std::string buf) {
  std::stringstream x;
  logger::get()->getBuffer(buf)->printAll(x);
  return x.str();
}
*/

File: python/Logger.py
import pyLogger

class Logger:
	def __init__(self):
	   """Initialize a logger object"""
		_log=pyLogger.get()

	def createFileBuffer(self,nm,fle,nflush=100):
		"""Create File buffer
		    nm- Name of the buffer
		    fle- Name of the file to write to
		    nflush - How often to flush the buffer"""
		    self._log.createFileBuffer(nm,fle,nflush)

	def createErrBuffer(self,nm,nflush=100):
		"""Create error buffer
		  nm- Name of the buffer
		  nflush - How often to flush"""
		self._log.createErrBuffer(nm,nflush)
	def flush(self):
		"""Flush all the buffers"""
		self._log.flush()
	def addMessage(self,buff,msg):
		"""Add a message to a given buffer"""
		self._log.addMessage(buff,msg)

	def get(self,buff):
		"""Get the contents of the specified buffer"""
		return self._log.get(buff)





log=Logger();

